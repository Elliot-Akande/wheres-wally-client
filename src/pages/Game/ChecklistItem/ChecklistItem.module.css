.image {
  max-height: 50px;
  max-width: 50px;
}

.listItem {
  padding: 8px 16px;

  display: flex;
  align-items: center;
  gap: 10px;

  background-color: var(--neutral-500);
  border-radius: 8px;
}

.listItemComplete {
  composes: listItem;

  opacity: 0.5;
  text-decoration: line-through;
}
