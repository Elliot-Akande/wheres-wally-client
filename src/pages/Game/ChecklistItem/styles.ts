import { css, styled } from "styled-components";

export const StyledChecklistItem = styled.li<{
  $checked: boolean
}>`
  padding: 8px 16px;

  display: flex;
  align-items: center;
  gap: 10px;

  background-color: var(--neutral-500);
  border-radius: 8px;
  transition: 0.3s;

  ${({ $checked }) =>
    $checked &&
    css`
      opacity: 0.5;
      text-decoration: line-through;
    `}
`;

export const imgStyles = css`
  max-height: 50px;
  max-width: 50px;
`;
