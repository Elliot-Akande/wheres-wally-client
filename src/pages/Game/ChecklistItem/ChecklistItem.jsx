import PropTypes from "prop-types";
import ImageLoader from "../../../components/ImageLoader/ImageLoader";
import styles from "./ChecklistItem.module.css";

const ChecklistItem = ({ character, found }) => {
  return (
    <li className={found ? styles.listItemComplete : styles.listItem}>
      <ImageLoader
        src={character.imageUrl}
        alt={character.name}
        imgClass={styles.image}
        spinnerColour={"hsl(288, 70%, 49%)"}
        spinnerWidth={3}
        spinnerHeight={26}
      />
      {character.name}
    </li>
  );
};

ChecklistItem.propTypes = {
  character: PropTypes.shape({
    name: PropTypes.string,
    imageUrl: PropTypes.string,
  }).isRequired,
  found: PropTypes.bool.isRequired,
};

export default ChecklistItem;
