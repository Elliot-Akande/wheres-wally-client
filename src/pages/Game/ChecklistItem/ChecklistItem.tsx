import ImageLoader from "../../../components/ImageLoader/ImageLoader";
import { StyledChecklistItem, imgStyles } from "./styles";

interface ChecklistItemProps {
  character: {
    name: string;
    imageUrl: string;
  };
  checked: boolean;
}

const ChecklistItem = ({ character, checked }: ChecklistItemProps) => {
  return (
    <StyledChecklistItem $checked={checked}>
      <ImageLoader
        src={character.imageUrl}
        alt={character.name}
        imgStyles={imgStyles}
        spinnerColour={"hsl(288, 70%, 49%)"}
        spinnerWidth={3}
        spinnerHeight={26}
      />
      {character.name}
    </StyledChecklistItem>
  );
};

export default ChecklistItem;
