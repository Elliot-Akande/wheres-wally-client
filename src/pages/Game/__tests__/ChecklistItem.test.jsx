import { describe, it, expect, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import ChecklistItem from "../ChecklistItem/ChecklistItem.jsx";

vi.mock("../../../components/ImageLoader/ImageLoader.jsx", () => ({
  default: ({ src }) => <img src={src}></img>,
}));

const character = {
  imageUrl: "#",
  name: "Aang",
};

describe("ChecklistItem", () => {
  it("renders correctly when found", () => {
    const { container } = render(
      <ChecklistItem character={character} found={true} />
    );

    expect(container).toMatchSnapshot();
  });

  it("renders correctly when not found", () => {
    const { container } = render(
      <ChecklistItem character={character} found={false} />
    );

    expect(container).toMatchSnapshot();
  });
});
