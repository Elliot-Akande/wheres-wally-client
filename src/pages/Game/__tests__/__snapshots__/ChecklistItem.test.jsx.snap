// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChecklistItem > renders correctly when found 1`] = `
<div>
  <li
    class="sc-beySPh gNxOao"
  >
    <img
      src="#"
    />
    Aang
  </li>
</div>
`;

exports[`ChecklistItem > renders correctly when not found 1`] = `
<div>
  <li
    class="sc-beySPh buzIbi"
  >
    <img
      src="#"
    />
    Aang
  </li>
</div>
`;
